doc/read-the-docs-site/howtos/exporting-a-blueprint.rst

```diff
@@ -0,0 +1,307 @@
+    { validatorTitle        :: Text
+    -- ^ A short and descriptive name for the validator.
+    , validatorDescription  :: Maybe Text
+    -- ^ An informative description of the validator.
+    , validatorRedeemer     :: ArgumentBlueprint referencedTypes
+    -- ^ A description of the redeemer format expected by this validator.
+    , validatorDatum        :: Maybe (ArgumentBlueprint referencedTypes)
+    -- ^ A description of the datum format expected by this validator.
+    , validatorParameters   :: Maybe (NonEmpty (ParameterBlueprint referencedTypes))
+    -- ^ A list of parameters required by the script.
+    , validatorCompiledCode :: Maybe ByteString
+    -- ^ A full compiled and CBOR-encoded serialized flat script.
+    }
+
+In our example this would be::
+
+  myValidator = MkValidatorBlueprint
+    { validatorTitle = "My Validator"
+    , validatorDescription = Just "An example validator"
+    , validatorParameters =
+        [ MkParameterBlueprint
+          { parameterTitle = Just "My Validator Parameters"
+          , parameterDescription = Just "Parameters configure the validator in compile-time"
+          , parameterPurpose = Set.singleton Spend
+          , parameterSchema = definitionRef @MyParams
+          }
+        ]
+    , validatorRedeemer =
+        MkArgumentBlueprint
+          { argumentTitle = Just "My Redeemer"
+          , argumentDescription = Just "A redeemer that does something awesome"
+          , argumentPurpose = Set.fromList [Spend, Mint]
+          , argumentSchema = definitionRef @MyRedeemer
+          }
+    , validatorDatum = Just
+        MkArgumentBlueprint
+          { argumentTitle = Just "My Datum"
+          , argumentDescription = Just "A datum that contains something awesome"
+          , argumentPurpose = Set.singleton Spend
+          , argumentSchema = definitionRef @MyDatum
+          }
+    , validatorCompiledCode = Nothing -- you can optionally provide the compiled code here
+    }
+
+The ``definitionRef`` function is used to reference a schema definition of a given type. It is 
+smart enough to discover the schema definition from the ``referencedType`` list and 
+fails to compile if the referenced type is not included.
+
+With all the pieces in place, we can now write the blueprint to a file::
+
+  writeBlueprint "/tmp/plutus.json" myContractBlueprint
+
+Annotations
+-----------
+
+Any `CIP-0057`_ blueprint type definition may include `optional keywords`_ to provide
+additional information:
+
+* title
+* description
+* $comment
+
+Its possible to add these keywords to a Blueprint type definition by annotating the
+Haskell type from which its derived with a corresponding annotation:
+
+* ``SchemaTitle`` 
+* ``SchemaDescription``
+* ``SchemaComment``
+
+For example, to add a title and description to the ``MyParams`` type,
+we can use the ``SchemaTitle`` and ``SchemaDescription`` annotations::
+
+  {-# ANN type MyParams (SchemaTitle "Title for the MyParams definition") #-}
```

Comment:
If you put the title in an annotation, then what do you put in the `parameterDescription` field? `Nothing`?

Reply:
Yes

Reply:
Ok, we should explain this in the doc (with an example would be great)

Reply:
I have misunderstood your question originally. The correct answer is that 
annotations are attached to constructors and then appear in the comment/titile/description fields in the schema definitions. The contract parameter and argument blocks have their own comment/title/description fields and those could be specified as literal values.

Reply:
I'm not sure I understand. Let's just use your example, in which you have:

```haskell
    , validatorParameters =
        [ MkParameterBlueprint
            { parameterTitle = Just "My Validator Parameters"
            , parameterDescription = Just "Compile-time validator parameters"
            , parameterPurpose = Set.singleton Spend
            , parameterSchema = definitionRef @MyParams
            }
        ]
```

But if you also have

```haskell
{-# ANN type MyParams (SchemaTitle "Title for the MyParams definition") #-}
```

then are the two titles, "My Validator Parameters" and "Title for the MyParams definition", both in the blueprint but in different places, or only one of them is chosen?

Reply:
Added paragraph about using `deriveParameterBlueprint` and `deriveArgumentBlueprint` TH functions.